src/bootstrap/src/core/build_steps/check.rs

```diff
@@ -158,6 +159,13 @@ pub struct Rustc {
     ///
     /// [`compile::Rustc`]: crate::core::build_steps::compile::Rustc
     crates: Vec<String>,
+    /// Whether to force `Kind::Check` on cargo invocations.
+    ///
+    /// By default, `Builder::kind` is propagated to the cargo invocations. However, there are cases
+    /// when this is not desirable. For example, when running `x clippy $tool_name`, passing `Builder::kind`
+    /// to cargo invocations would run clippy on the entire compiler and library, which is not useful if
+    /// we only want to lint a few crates with specific rules.
+    force_check: bool,
```

Comment:
Hm, I guess this feels a little off to me. I'd expect this to be something closer to `kind: Kind` perhaps? Otherwise we could run twice (e.g., because you have a `x.py check` run that ends up wanting to do something that passes `force_check: true` here.

I suspect that's somewhat rare but in general I'd try to stay away from "force" options like this and instead specify what we are requesting to have happen.

Reply:
Sure, [changed](https://github.com/rust-lang/rust/pull/128269/commits/92ca0a6a04c65cd8b23b09c689fc73b5bf2cfc58) the `force_check` logic with `override_build_kind`.